### DISCUSSION TITLE
Consider differentiating paths of autocoded files

### DISCUSSION THREAD

**Discussion by thomas-bc** [nasa/fprime#2627] [Call for Comments]
| | |
|:---|:---|
|**_F´ Version_**| 3.3 |
|**_Affected Component_**| Autocoded files |
---
## Feature Description

Autocoded files have historically been output in the build cache under a file structure that mirrors the source file structure. This is really convenient for browsing the build-cache, but also comes with some cost.
e.g. [here](https://github.com/fprime-community/fprime-workshop-led-blinker/blob/cc047ca3ba5dc941d18fd497533a71e011c2c89e/LedBlinker/Top/LedBlinkerTopology.cpp#L8C1-L9C51)
```
#include <LedBlinker/Top/LedBlinkerPacketsAc.hpp>
#include <LedBlinker/Top/LedBlinkerTopologyAc.hpp>
```
To an avid user, it naturally becomes pretty clear that those will be in the build cache (because of the `Ac` suffix), but this pattern can confuse less experimented users. And even for experimented developers, it takes an extra couple seconds to read each suffixes in logs etc... when issues arise. My point is, the differentiation between source and auto-coded files could be made clearer.

I am proposing we consider moving to a structure where file paths would be mirrored, but by some mean differentiated.
Examples could be:
```
#include <AC/LedBlinker/Top/LedBlinkerPacketsAc.hpp>
``` 
```
#include <Autocoded/LedBlinker/Top/LedBlinkerPacketsAc.hpp>
```
Or please make suggestions on how this should/could be achieved!

## Rationale

Differentiate between autocoded and source files more easily.

**Comment by LeStarch** [comment]
I am going to convert this to a discussion, as it mentions "consider".

**Comment by LeStarch** [comment]
I really like this idea, as it makes the code considerably more clear.  It also helps the coverage program (which has to do this internally).

However, this may be difficult from a build-system organization perspective.  It also means substantial changes on every component across all F´.  It would be the "breaking change".  @timcanham and @bocchino what are your thoughts here.  Would this change be worth the cross-cutting impact?
